compiler/src/parsing/parsetree.re

```diff
@@ -327,7 +327,8 @@ type prim0 =
   | AllocateFloat64
   | AllocateRational
   | WasmMemorySize
-  | Unreachable;
+  | Unreachable
+  | HeapStart;
```

Comment:
It looks like we didn't actually need the changes to `HeapStart`, can we back those out?

Reply:
How do you mean? How would you query where the start of the heap is then?

Reply:
Oh not that, just that you changed the primitive to be a prim0 instead of a prim_constant, but I don't see anywhere why it needs to be a function.

Reply:
Oh ok gotcha; in order to fit in with [the rest of the translprim code](https://github.com/grain-lang/grain/blob/242d845a91b0bb7769f7fc4b2a1ad7dfec41eec1/compiler/src/typed/translprim.re#L1503), should I create a new parsetree node like `PExpHeapBase` to be only used in this context?

Reply:
Nah. We kinda hate this pattern but haven't gotten around to changing it.

Reply:
Ok how would you suggest that code segment be changed then?

Reply:
Just putting it back to the way it was—we didn't have this variant defined here because there aren't any constant prims in the parsetree.

Reply:
Unless I'm missing something, it wouldn't work going back to the way it was would it? Because the heap start is dependent on the size of the table, which isn't known until link time, you can't know what number to plug in here when compiling a module

Reply:
Yeah, sorry for not being clear. Not the functionality being the same as it was before; just the primitive continuing to be a prim_constant, but the value is pulled from the imported global in `compcore` (instead of being set in `linearize` like it was before). Does that make sense?

Reply:
Sorry, I still don't quite understand what you mean. If `HeapStart` is added back to `primitive_constant` that would necessitate the value being supplied for `@heap.start` [here](https://github.com/grain-lang/grain/blob/242d845a91b0bb7769f7fc4b2a1ad7dfec41eec1/compiler/src/typed/translprim.re#L1500). Without significant changes to how this segment of code in `translprim`, I'm not sure how you would do that. And the only thing you could really do in `compcore` w.r.t. getting a reference to the global is to create a `Global_get` binaryen `Expression` right? The current code is `translprim` is not set up to handle that and instead expects `Parsetree.expression`s

Reply:
I see what you mean. That's... annoying. I suppose we can keep it the way it is for now 😩

Reply:
Yeah it is a bit annoying haha. Maybe an alternative here could be to move `prim_constant` into the parsetree and then handle compiling them in `compcore` in the same manner as the other `prim`s; that would require a bit of extra work but would add some consistency and also allow `@heap.start` to be a constant :shrug:

Reply:
I realized that, but it makes me sad.

Reply:
I'll leave it up to you.

Reply:
Lol ok I think it's fine to have it as a prim0 for now; it could always be changed in 0.7 or later anyways. It doesn't look like this primitive gets used anywhere in grain stdlib or runtime code right now anyways